server/src/main/java/org/elasticsearch/snapshots/RestoreService.java

```diff
@@ -988,6 +991,13 @@ private static void validateSnapshotRestorable(final RepositoryMetadata reposito
         }
     }
 
+    static void validateGlobalStateRestorable(RestoreSnapshotRequest request, Snapshot snapshot, SnapshotInfo snapshotInfo) {
+        if (request.includeGlobalState() && snapshotInfo.includeGlobalState() != Boolean.TRUE) {
+            request.includeGlobalState(false);
+            logger.warn("[{}] the snapshot was created without global state, skipping restoring global sate", snapshot);
```

Comment:
Maybe word this explicitly and say "[{}] was created without global state but restore request [{}] asks for global state restore explicitly, skipping global state restore"